### PR TITLE
Use a more consistent widget when adding/renaming layer style

### PR DESCRIPTION
**Pull request by DelazJ**
Currently, when trying to add a style whose name already exists, the creation simply silently fails. No feedback. Instead of an input dialog, let's rely  on the qgsnewnamedialog widget.

https://user-images.githubusercontent.com/7983394/147813082-1843476a-63ca-4ac8-85e9-8428eafcc89a.mp4

### CODE REVIEW

File: src/gui/qgsmaplayerstyleguiutils.cpp [hunk 1]
```diff
@@ -130,15 +132,27 @@ void QgsMapLayerStyleGuiUtils::addStyle()
...
     return;
 
-  bool ok;
-  const QString text = QInputDialog::getText( nullptr, tr( "New Style" ),
-                       tr( "Style name:" ), QLineEdit::Normal,
-                       QStringLiteral( "new style" ), &ok );
-  if ( !ok || text.isEmpty() )
+  const QStringList styles = layer->styleManager()->styles();
+
+  QgsNewNameDialog dlg( nullptr, tr( "new style" ), QStringList(), styles, Qt::CaseSensitive );
```
Comment (DelazJ) [inline src/gui/qgsmaplayerstyleguiutils.cpp:137]: I don't know what to use as parent parameter (in both contexts), probably why the widget is not centered on the dialog. Help, welcome.

Comment (elpaso) [reply src/gui/qgsmaplayerstyleguiutils.cpp:137]: Tricky: the properties dialog is very far in the call chain the invoked this code. You may try to retrieve it from the menu parent. f I get this correctly the action `a` parent is the menu and the menu's parent is the properties dialog.

Comment (m-kuhn) [reply src/gui/qgsmaplayerstyleguiutils.cpp:137]: - change the `parent` parameter of `actionAddStyle` to `QMenu *` 
- move the code from `addStyle` to a lambda (i.e. line 39: `connect( a, &QAction::triggered, this, [a, layer, menu]() {...} ] );`)
- Nice added benefit, we can now drop the `qobject_cast` code at the beginning of this method and use `menu->parent()` for the dialog and the `setData()` call on line 38 also

Comment (m-kuhn) [reply src/gui/qgsmaplayerstyleguiutils.cpp:137]: @DelazJ does this make sense to you?

Comment (DelazJ) [reply src/gui/qgsmaplayerstyleguiutils.cpp:137]: I think, in the "big" lines 😄 . But I want to test locally that I really understand and not use the CI to test and disturb any of you with my attempts. Will try to get back to this ASAP. Thanks for the hints.

Comment (DelazJ) [reply src/gui/qgsmaplayerstyleguiutils.cpp:137]: I Made what I understood from the suggestion above, built with no error but when calling the tool, no dialog appeared. Then I tried some changes to "fix" this but failed to build 😢

Comment (m-kuhn) [reply src/gui/qgsmaplayerstyleguiutils.cpp:137]: Can you push the current version? It will be easier to comment.

**Pull request closed without merging**